statement.py

```diff
@@ -4,43 +4,396 @@
+                lines = list(lines)
+                key = dict(key)
+                move = origin.statement._get_move(key)
+                moves.append((move, lines))
+
+        Move.save([m for m, _ in moves])
+
+        to_write = []
+        for move, lines in moves:
+            to_write.extend((lines, {'move': move.id}))
+        if to_write:
+            StatementLine.write(*to_write)
+
+        move_lines = []
+        for move, lines in moves:
+            amount = 0
+            amount_second_currency = 0
+            for line in lines:
+                move_line = line.get_move_line()
+                if not move_line:
+                    continue
+                move_line.move = move
+                amount += move_line.debit - move_line.credit
+                if move_line.amount_second_currency:
+                    amount_second_currency += move_line.amount_second_currency
+                move_lines.append((move_line, line))
+
+            move_line = origin.statement._get_move_line(
+                amount, amount_second_currency, lines)
+            move_line.move = move
+            move_lines.append((move_line, None))
+
+        MoveLine.save([l for l, _ in move_lines])
+        StatementLine.reconcile(move_lines)
+        return moves
+
+    @classmethod
+    @ModelView.button
+    @Workflow.transition('registered')
+    def register(cls, origins):
+        pass
 
-    #TODO: check the button, to see if it works correctly
     @classmethod
     @ModelView.button
-    def reconcile(cls, origins):
+    @Workflow.transition('posted')
+    def post(cls, origins):
         pool = Pool()
-        Line = pool.get('account.statement.line')
+        Statement = pool.get('account.statement')
+        StatementLine = pool.get('account.statement.line')
+        Move = pool.get('account.move')
+        MoveLine = pool.get('account.move.line')
+
+        cls.validate_origin(origins)
+        cls.create_moves(origins)
+
+        lines = [l for o in origins for l in o.lines]
+        # It's an awfull hack to sate the state, but it's needed to ensure the
+        # Warning of statement state in Move.post is not applied when try to
+        # concile and individual origin. For this, need the state == 'posted'.
+        statements = [o.statement for o in origins]
+        statement_state = []
+        for origin in origins:
+            statement_state.append([origin.statement])
+            statement_state.append({
+                    'state': origin.statement.state,
+                    })
+        if statements:
+            Statement.write(statements, {'state': 'posted'})
+        StatementLine.post_move(lines)
+        if statement_state:
+            Statement.write(*statement_state)
+        # End awful hack
+
+        statements = []
+        for origin in origins:
+            statement = origin.statement
+            try:
+                getattr(statement, 'validate_%s' % statement.validation)()
```

Comment:
I think it would be cleaner to create a new validation type that would be the one used in enable banking statements. So no need for this kind of validation.

Reply:
As we discussed, I've tried to do the minimum of new code. But I agree that will be better to have an origin validation type.

Reply:
I think is better to use the core validations to ensure the account statements are correct to 'post' them.